test/IECoreScene/FontTest.py

```diff
@@ -107,5 +107,12 @@ def testLineSpacing( self ) :
 
 		self.assertGreater( font.bound( "T\nT" ).size().y, font.bound( "TT" ).size().y )
 
+	def testInvalidChar( self ) :
+
+		font = IECoreScene.Font( os.path.join( "test", "IECore", "data", "fonts", "Vera.ttf" ) )
+
+		self.assertEqual( font.bound( "XX" ), font.bound( "\xff" ) )
```

Comment:
There's a test failure here :

```AssertionError: Box2f(V2f(0.0297851562, 0), V2f(1.33886719, 0.729003906)) != Box2f(V2f(0.0297851562, 0), V2f(0.653808594, 0.729003906))```

I think this is just because we have two characters on the left and one on the right? Although that wouldn't explain why the test is passing on Windows but not Linux...

Reply:
The reason this is passing on Windows is because Windows somewhere is decoding the `\xff` character, which is `ÿ` in UTF-8, as `Ã¿`. That's the representation of `\xff` when it is decoded using `cp1252` which is the default encoding set by Windows locale.

You can demonstrate this on Linux with `"\xff".encode().decode("cp1252")`

There's some more information here : https://www.i18nqa.com/debug/utf8-debug.html

Where exactly this decoding is happening I haven't tracked down yet, the c++ code in IECoreScene/Font.cpp is definitely getting called with `Ã¿` as its input.

So I don't have a solution for Windows yet...

Reply:
Testing on Windows and my local Linux installation I get a value of `-1` in the c++ code when using `font.bound( b"\xff" )`, so testing it against a single `"X"` passes.

Reply:
Updated test with this change, worked on my Linux, if it passes CI, I guess we're good.